internal/collector/processor/multi_processor_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"fmt"
+	"testing"
+
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+func TestMultiSpanProcessorMultiplexing(t *testing.T) {
+	processors := make([]SpanProcessor, 3)
+	for i := range processors {
+		processors[i] = &mockSpanProcessor{}
+	}
+
+	tt := NewMultiSpanProcessor(processors...)
+	batch := &agenttracepb.ExportTraceServiceRequest{
+		Spans: make([]*tracepb.Span, 7),
+	}
+
+	var wantSpansCount = 0
+	for i := 0; i < 2; i++ {
+		wantSpansCount += len(batch.Spans)
+		tt.ProcessSpans(batch, "test")
+	}
+
+	for _, p := range processors {
+		m := p.(*mockSpanProcessor)
+		if m.TotalSpans != wantSpansCount {
```

Comment:
The type being tested is just a fanout to other span processors, it doesn't drop any nil span. The mock used here just counts the length.

Concurrency protection is not needed since no state is stored, everything is local to the function call.

Reply:
A good and useful convention will be to require that receivers never pass nil nodes. Anyway, I will add empty spans to avoid any confusion here.